spec/lib/salus/scanners/bandit_spec.rb

```diff
@@ -0,0 +1,437 @@
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        expect(scanner.report.passed?).to eq(false)
+
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        filenames = logs['results'].map { |r| r['filename'] }.uniq
+        expect(filenames).to eq(%w[./main.py])
+      end
+
+      it 'ini file option should work' do
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns2")
+
+        # ini_file.yaml points to file that specifies exclude main2.py
+        config_file = "#{py_dir}/salus_configs/ini_file.yaml"
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        expect(scanner.report.passed?).to eq(false)
+
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        expect(logs['results'].size).to eq(1)
+        expect(logs['results'][0]['filename']).to eq('./main.py')
+      end
+
+      it 'nosec option should work' do
+        # repo has a nosec comment
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns3")
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: {})
+        scanner.run
+        expect(scanner.report.passed?).to eq(true)
+
+        # yaml specifies ignore-nosec
+        config_file = "#{py_dir}/salus_configs/ignore_nosec.yaml"
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        expect(scanner.report.passed?).to eq(false)
+
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        results = logs['results']
+
+        expect(results.size).to eq(1)
+        expect(results[0]['test_id']).to eq('B105')
+      end
+
+      it 'exclude path option should work' do
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns2")
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: {})
+        scanner.run
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        confidence = logs['metrics']['_totals']
+
+        expect(scanner.report.passed?).to eq(false)
+        expect(confidence['CONFIDENCE.HIGH']).to eq(5)
+        expect(confidence['CONFIDENCE.LOW']).to eq(0)
+        expect(confidence['CONFIDENCE.MEDIUM']).to eq(2)
+        expect(confidence['CONFIDENCE.UNDEFINED']).to eq(0)
+
+        # exclude 1 file
+        config_file = "#{py_dir}/salus_configs/exclude1.yaml"
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        confidence = logs['metrics']['_totals']
+
+        expect(scanner.report.passed?).to eq(false)
```

Comment:
How are these confidence metrics related to excluding paths?

Reply:
I was checking if number of vulnerabilities found per confidence level decreased as I excluded paths.  I agree this was not ideal, so I updated the code to check for actual file names in the results instead.